light-client/starknet-storage-verifier/src/converter/value.rs

```diff
@@ -0,0 +1,54 @@
+use std::str::FromStr;
+
+use ibc_core::host::types::identifiers::{ChannelId, PortId};
+use ibc_core::host::types::path::Path;
+use poseidon::Poseidon3Hasher;
+use starknet_core::utils::starknet_keccak;
+use starknet_crypto::{pedersen_hash, Felt};
+
+pub fn convert_value(value: &[u8]) -> Felt {
+    starknet_keccak(value)
+}
+
+pub fn convert_storage_value(path: &str, storage_member: &Felt) -> Felt {
+    let path = Path::from_str(path).unwrap();
+    match path {
+        Path::SeqAck(seq_ack_path) => {
+            let next_sequence_ack_key = next_sequence_ack_key(seq_ack_path.0, seq_ack_path.1);
+
+            let path_sn_keccak = starknet_keccak(&next_sequence_ack_key.to_bytes_be());
+            pedersen_hash(&path_sn_keccak, storage_member)
+        }
+        _ => unimplemented!(),
+    }
+}
+
+pub fn next_sequence_ack_key(port_id: PortId, channel_id: ChannelId) -> Felt {
+    let mut raw_path: Vec<Felt> = vec![];
+    raw_path.extend(serialize_byte_array("nextSequenceAck".as_bytes()));
+    raw_path.extend(serialize_byte_array("ports".as_bytes()));
+    raw_path.extend(serialize_byte_array(port_id.as_bytes()));
+    raw_path.extend(serialize_byte_array("channels".as_bytes()));
+    raw_path.extend(serialize_byte_array(channel_id.as_bytes()));
```

Comment:
can we use the cairo encoding here ? probably not, as it creates dependency to relayer workspace 😞

Reply:
You are right, we would need to move it